File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int zero = 0;
        int one = 0;
        for (char c: s.toCharArray())
            if (c == '1')
                one += 1;
        
        int result = 0;
        
        for (int i = 0; i < s.length() - 1; i++)
        {
            if (s.charAt(i) == '0')
            {
                zero += 1;
            }
            else{
                one -= 1;
            }
            result = Math.max(result, zero + one);
        }

        return result;
    }
}

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        l = [0, 0]
        r = [0, 0]

        for i in s:
            r[int(i)] += 1

        result = 0
        for i in s[:-1]:
            l[int(i)] += 1 
            r[int(i)] -= 1
            result = max(result, l[0] + r[1])

        return result